[core] Fixed wrong check of common FEC configuration (#1811)
Haivision/srt: docs/APISocketOptions.md

Set up the packet filter. The string must match appropriate syntax for packet
filter setup.

Empty value of this option means that for this connection party filter isn't
required, but it will accept any filter settings if provided by the peer. If
this option is changed in both parties simultaneously, the resulting
configuration will be an integrated configuration from both parties, that is,
parameters that are set only on one side will have the value defined by that
side, and parameters not set in either side will be set as default. In the

> following cases:

Reviewer: ```suggestion
An empty value for this option means that for this connection the filter isn't
required, but it will accept any filter settings if provided by the peer. If
this option is changed by both parties simultaneously, the result will be a configuration integrating parameters from both parties (that is, parameters that are set only on one side will have the value defined by that
side, and parameters not set in either side will be set as default). 

In the following cases:
```

Set up the packet filter. The string must match appropriate syntax for packet
filter setup.

An empty value for this option means that for this connection the filter isn't
required, but it will accept any filter settings if provided by the peer. If
this option is changed by both parties simultaneously, the result will be a
configuration integrating parameters from both parties, that is:

* parameters provided by both parties are accepted, if they are identical
* parameters that are set only on one side will have the value defined by that side
* parameters not set in either side will be set as default

> In the following cases:
> 
> * both sides define a different packet filter type
> * for the same key two different values were provided by both sides
> * mandatory parameters weren't provided by either side
> 
> the connection will be rejected with `SRT_REJ_FILTER` code.

Reviewer 2: ```suggestion
The connection will be rejected with `SRT_REJ_FILTER` reason code In the following cases:

* both sides define a different packet filter type;
* for the same key two different values were provided by both sides (e.g. `cols:10` and `cols:12`);
* mandatory parameters weren't provided by either side.

```

In case of the built-in `fec` filter, the mandatory parameter is `cols`, all
others have their default values. For example, the configuration specified

> as `fec,cols:10` is `fec,rows:1,cols:10,arq:onreq,layout:even`.

Reviewer 2: A link to [Configuring the FEC Filter](packet-filtering-and-fec.md#configuring-the-fec-filter) would be useful here, to easily find those configuration parameters.

> Examples for the built-in `fec` filter (in "negotiated config" the parameters
> with default values are skipped):

Reviewer 2: ```suggestion
Examples for the built-in `fec` filter **(in "negotiated config" the parameters
with default values are skipped**):
```

| Peer A               | Peer B      | Negotiated Config            | Result                   |
|----------------------|-------------|------------------------------|--------------------------|
| (no filter)          | (no filter) | (no filter)                  | OK                       |
| fec                  | (no filter) | fec                          | missing `cols` parameter |
| fec,cols:10          | fec         | fec,cols:10                  | OK                       |
| FEC,cols:10          | FEC,cols:10 | FEC,cols:10                  | unknown filter 'FEC'     |
| fec,layout:staircase | fec,cols:10 | fec,cols:10,layout:staircase | OK                       |
| fec,cols:20          | fec,cols:10 | fec                          | parameter value conflict |

In general it is recommended that one party defines the full configuration,
while the other keeps this value empty.

> If you read this option after the connection is established, it will return
> the full configuration that has been agreed upon by both parties (including
> default values).

Reviewer 2: ```suggestion
Reading this option after the connection is established will return
the full configuration that has been agreed upon by both parties (including
default values).
```

the connection will be rejected with `SRT_REJ_FILTER` code.

In case of the builtin `fec` filter, the mandatory parameter is `cols`, all
others have their default values. For example, the configuration specified
as `fec,cols:10` is `fec,rows:1,cols:10,arq:onreq,layout:even`.

> Examples for the builtin `fec` filter:

Reviewer 2: built-in

Author: Then this should be better changed across the docs.
```
> grep builtin *.md
packet-filtering-and-fec.md:their own filters. SRT provides also one builtin filter named "fec". This
packet-filtering-and-fec.md:* alter the packet before inserting it into the SRT channel (the builtin "fec"
```

| Peer A               | Peer B      | Negotiated Config            | Result                   |
|----------------------|-------------|------------------------------|--------------------------|
| (no filter)          | (no filter) | (no filter)                  | OK                       |
| fec                  | (no filter) | fec                          | missing `cols` parameter |

> | fec,cols:10          | fec         | fec,cols:10                  | OK                       |

Reviewer 2: Negotiated: `fec,cols:10,rows:1,layout:even`

Author: I wanted to add an explanation that default values are skipped in this final configuration.

Actually negotiated is: `fec,cols:10,rows:1,layout:even,arq:onreq`.

Reviewer 2: This negotiated value is what is expected in \SRTO_PACKETFILTER\ after a connection is established.

Author: Yes, and that's the case. The test application has been added therefore an extra procedure to display that option after connection. Parameters are maybe not in that order, but all parameters are present.
